### ISSUE TITLE
Qt client: a broken possibility for change status message function

### ISSUE THREAD

**Issue by amirmahdifard** [BearWare/TeamTalk5#2100]
### Description

it's a possibility of teamtalk that clients can change nicknames and status messages befor they can connect to any server, emajin, you are currently connected to a server, with a status message: and to morro when you open your teamtalk, you wish to delete your status message or change it before you even log in to a server, as you can change your nickname before connecting to a server: but if you wish to change your status message before connecting to any server, you must edit your teamtalk5.ini file and do it from there: this is resolved so easyly: please, open a pull request, and just remove this line of code from the mainwindow.cpp
    ui.actionChangeStatus->setEnabled(auth);
then everything will be resolved: as @bear101 talked about this kind of things before, i think he is also agreed because he knows this: thanks! @CoBC can i please ask you to do this and open this pull request and we'll see what happens next? thanks much! i know coding but i don't know how to open pull requests yet, but i'm working to lurn: so can i please ask you to do this? i never ask for something unwanted: thanks!

### Application

- [X] qtTeamTalk
- [ ] TeamTalkAndroid
- [ ] iTeamTalk
- [ ] TeamTalkServer

**Comment by CoBC** [comment]
You already asked it some month ago, we heard your request.

**Comment by amirmahdifard** [comment]
@CoBC yes but in that time i didn't know the code and that issue got closed just because no one opened any pull request: so cab you please open the pull request this time and we will see if it gets confurmed? please, thanks to you.

**Comment by amirmahdifard** [comment]
if i could, i would open pull request my self and would see if it would get murged as i'm asking for something that makes sence, just idk why github doesn't let me ad this repo, i think i don't have that right, and i'm having some problems with it, thanks.

**Comment by amirmahdifard** [comment]
@CoBC if i want to be able to open pull requests i should creat a fork of this repo in to my account: correct?

**Comment by CoBC** [comment]
Yes

**Comment by CoBC** [comment]
Note, however, the line you mensioned previously isn't the only one thing to change to reply to your request, the action to change status call a library function (TT_DoChangeStatus) when used, so, if you call this function without be logged in to a server, this will fail, so you have to change behaviour of slot change status and check if the client is authentified or not to the server and adapt the code according to this.

**Comment by amirmahdifard** [comment]
do i check the box that says copy master branch?

**Comment by CoBC** [comment]
Just let default options

**Comment by amirmahdifard** [comment]
yes i understand, but i just want to remove this so it doesn't enable this option only if the client is auth wich means authorised and connected as i see what auth does in the bool auth: right?

**Comment by amirmahdifard** [comment]
ok i did this: now i choose bace repo as amirmahdifard and head repo as bearware: right?

**Comment by CoBC** [comment]
Yes, but no, as I said previously, you can't change it just with removal of bool auth in this line.
And yes, you're right for fork, anyway I think it's not the apropriate way to talk about it, open a discussion, read GItHub doc or go in private to talk about fork, PR ant other.

**Comment by amirmahdifard** [comment]
@CoBC ok: so will you open a pull request for this time, or no? if no, i will close this and reopen it whenever i can do it: or if you can, i ask you to do this please, however, you can answer freely ahat will you do about this so i know it too: thanks!

**Comment by amirmahdifard** [comment]
@CoBC do you mean this part of the code? i found this that only allowes tt do change status if the client is authorised, but this is related to the toggle question mode option.
void MainWindow::slotToggleQuestionMode(bool checked)
{
    if(checked)
        m_statusmode |= STATUSMODE_QUESTION;
    else
        m_statusmode &= ~STATUSMODE_QUESTION;

    QString statusmsg = ttSettings->value(SETTINGS_GENERAL_STATUSMESSAGE).toString();
    if(TT_GetFlags(ttInst) & CLIENT_AUTHORIZED)
        TT_DoChangeStatus(ttInst, m_statusmode, _W(statusmsg));
}

**Comment by amirmahdifard** [comment]
there's also a fue more but they are related to the idle timers and stuf, they are not about this option in the me menu.
